### DISCUSSION TITLE
RealityData Attached

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#5753] [Q&A] [answered]
Hi all, I have a problem with a RealityData , which I can't turn off and on.

when i go to benteley app "https://connect.bentley.com/"
Itwin services, iModel Management in the realityData section this reality is written "(attached)" and I would not like this to be the reason why it is not turned on or off, i.e. the fact that it is Attached to the iModel. I am attaching a small photo of the item.

thanks for the precious help
<img width="319" alt="ortofoto_1" src="https://github.com/iTwin/itwinjs-core/assets/57446652/1c6b124f-4ea1-4bf1-97e0-3ad58e73a1d1">

**Comment by pmconne** [comment]
I believe that "(attached)" means that the iModel contains a [geometric model](https://www.itwinjs.org/reference/core-backend/models/geometricmodel3d/) representing the reality data. Its json properties will contain a URL pointing to the reality data. These kinds of models are produced, for example, by the MicroStation connector when it processes a reality model reference attachment like a scalable mesh or point cloud.

In contrast to "attached" reality models, ["context" reality models](https://www.itwinjs.org/reference/core-common/displaystyles/contextrealitymodel/) can be freely attached and detached from a view at display time.

Can you clarify what you mean by "not turned on or off"? You should be able to toggle its visibility in the viewport via the eye icon to the left of its name. I don't think you can remove it from the list of available reality models due to it being contained in the iModel.

**Comment by Paolo-lab** [comment]
hello good morning thanks for the reply.

I have a web app that turns my reality shows on and off, which works fine on all my reality shows except this one D1AZ_13_Ortofoto3D, which seems to remain attached to my model, so I went to https://connect.bentley.com/ to see how it was managed and since it's the only reality show with "attached" written, I doubted that it was due to this.

**Reply by pmconne** [reply]
If you are in need of further assistance, please specify what problems remain.

**Reply by Paolo-lab** [reply]
hello, sorry maybe I'm not explaining myself, yes the problem is the same, I have this realitydata "D1AZ_13_Ortofoto3D" that I can't remove from my imodel, so the question is does it depend on whether it is attached ?
because if so I will have to ask to reload the reality show without Attached

**Reply by pmconne** [reply]
I explained what "attached" vs "context" reality models are. If the reality model is persisted inside your iModel ("attached") then the only way to "remove" it is to delete the model; or rerun the connector on the source files without the reality data reference attachment.

Can you be clearer about why you want to "remove" this reality data?

I don't understand what "reload the reality show" means.

**Reply by Paolo-lab** [reply, marked as answer]
Non capisco cosa significhi "ricaricare il reality show".

i meant reload iModel without realitydata "attached"

Forgive me but I didn't understand

I explained what "attached" vs "context" reality models are. If the reality model is persisted inside your iModel ("attached") then the only way to "remove" it is to delete the model; or rerun the connector on the source files without the reality data reference attachment.


The end customer in the testing phase disputes that he is unable to remove all the reality data (he would like to see the artifact only without background maps, nothing in short) but he is left with this reality data, we are talking about the one mentioned in the chat.
perhaps he is not aware of the "attached" models of reality either.
so now there's a reason and I know how to respond.

I thank you, now everything is clear to me, sorry if I made you waste some time

Thank you
Paul
